### PR TITLE
getEditsForFileRename: Test both before and after the rename

### LINKED ISSUE #25059: Incorrect getEditsForFileRename with npm linked module
From https://github.com/Microsoft/vscode/issues/52022

**TypeScript Version:**  3.0.0-dev.20180616

**Search Terms:** 

**Code**
1. `git clone https://github.com/mjbvz/ts-update-paths-npm-link-bug`
1. `npm install`
1. Open project in VS Code
1. Open file `d.ts`
1. Rename `d.ts` to `a.ts`

**Expected behavior:**
No edits returned from `getEditsForFileRename `

**Actual behavior:**
`getEditsForFileRename` changes the import for `import { abc } from "sub";` to `import { abc } from "./sub/src/types"`

```
[Trace  - 2:34:49 PM] Response received: getEditsForFileRename (52). Request took 8 ms. Success: true 
Result: [
    {
        "fileName": "/Users/matb/projects/san/a.js",
        "textChanges": [
            {
                "start": {
                    "line": 3,
                    "offset": 22
                },
                "end": {
                    "line": 3,
                    "offset": 25
                },
                "newText": "./sub/src/types"
            }
        ]
    }
]
```
**Playground Link:**  

**Related Issues:**

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #25059 and #25029

Updates the fourslash tests to run once before the rename, perform the rename, then run again. The bug in #25059 only reproduced if `getEditsForFileRename` was called *after* the rename, which we weren't testing before.

### CODE REVIEW

File: src/compiler/moduleSpecifiers.ts [hunk 1]
```diff
@@ -2,13 +2,23 @@
...
     export interface ModuleSpecifierPreferences {
-        importModuleSpecifierPreference?: "relative" | "non-relative";
+        readonly importModuleSpecifierPreference?: "relative" | "non-relative";
     }
 
-    // Note: fromSourceFile is just for usesJsExtensionOnImports
-    export function getModuleSpecifier(compilerOptions: CompilerOptions, fromSourceFile: SourceFile, fromSourceFileName: string, toFileName: string, host: ModuleSpecifierResolutionHost, preferences: ModuleSpecifierPreferences = {}) {
-        const info = getInfo(compilerOptions, fromSourceFile, fromSourceFileName, host);
-        return getGlobalModuleSpecifier(toFileName, info, host, compilerOptions) ||
+    // Note: importingSourceFile is just for usesJsExtensionOnImports
```
Comment (mhegazy) [inline src/compiler/moduleSpecifiers.ts]: should that instead be a boolean flag?

File: tests/cases/fourslash/getEditsForFileRename_tsconfig.ts [hunk 2]
```diff
@@ -16,6 +16,9 @@
 ////    "exclude": ["old"],
 ////}
 
+// @Filename: /src/old/someFile.ts
+////
```
Comment (sheetalkamat) [inline tests/cases/fourslash/getEditsForFileRename_tsconfig.ts]: With this change it seems we need to have oldFile present on the disk? What about the tests when getEditsForRename when oldFile isnt present on the disk?

Comment (andy-ms) [reply tests/cases/fourslash/getEditsForFileRename_tsconfig.ts]: For `getEditsForFileRename` to work, either oldFile xor newFile should be present. The tests start with oldFile present, verify the output of `getEditsForFileRename`, then rename the file, and verify the output of `getEditsForFileRename` again.

File: src/compiler/moduleSpecifiers.ts (hunk 1 above)
Comment (andy-ms) [reply src/compiler/moduleSpecifiers.ts]: Let's do that in #25073 which is making this configurable.

**Pull request merged**